sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1767,6 +1767,11 @@ object SQLConf {
         "with String")
     .booleanConf
     .createWithDefault(false)
+
+  val DEFAULT_V2_CATALOG = buildConf("spark.sql.default.catalog")
```

Comment:
nit: `spark.sql.catalog.default` is more consistent with other SQL config names: `spark.sql.componentName.featureName`.

Reply:
The reason for this order is that `spark.sql.catalog.(name)` properties are used to register catalogs. So the property name you suggest creates a catalog named "default" using the existing convention. We could change the properties to use `catalogs` instead if you'd prefer. Then the `catalog.default` name would not conflict.

Reply:
As @rdblue mentioned, `spark.sql.catalog.default` has meaning. Also, the `spark.sql.default.catalog` was the name already used in our code base before this PR.

Reply:
Ah I see the problem now.

Since we already have `spark.sql.defaultSizeInBytes`, shall we name it `spark.sql.defaultCatalog`?

Reply:
That works for me. I'll open a PR to rename.

Reply:
+1, too~